frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseCheckoutAction.java

```diff
@@ -0,0 +1,131 @@
+import io.vavr.control.Option;
+import kr.pe.kwonnam.slf4jlambda.LambdaLogger;
+import lombok.CustomLog;
+import lombok.NonNull;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeySelectedBranchName;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.gitmachete.frontend.vfsutils.GitVfsUtils;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@ExtensionMethod({GitVfsUtils.class, GitMacheteBundle.class})
+@CustomLog
+public abstract class BaseCheckoutAction extends BaseGitMacheteRepositoryReadyAction
+    implements
+      IExpectsKeySelectedBranchName {
+
+  @Override
+  public LambdaLogger log() {
+    return LOG;
+  }
+
+  protected abstract @Nullable String getTargetBranchName(AnActionEvent anActionEvent);
+
+  @Override
+  @UIEffect
+  protected void onUpdate(AnActionEvent anActionEvent) {
+    super.onUpdate(anActionEvent);
+
+    val presentation = anActionEvent.getPresentation();
+    if (!presentation.isEnabledAndVisible()) {
+      return;
+    }
+
+    val targetBranchName = getTargetBranchName(anActionEvent);
+
+    // It's very unlikely that targetBranchName is empty at this point since it's assigned directly before invoking this
+    // action in EnhancedGraphTable.EnhancedGraphTableMouseAdapter#mouseClicked; still, it's better to be safe.
+    if (targetBranchName == null || targetBranchName.isEmpty()) {
+      presentation.setEnabled(false);
+      presentation.setDescription(getNonHtmlString("action.GitMachete.BaseCheckoutAction.undefined.branch-name"));
+      return;
+    }
+
+    val currentBranchName = getCurrentBranchNameIfManaged(anActionEvent);
+
+    if (currentBranchName != null && currentBranchName.equals(targetBranchName)) {
+      presentation.setEnabled(false);
+      presentation.setDescription(
+          getNonHtmlString("action.GitMachete.CheckoutSelectedAction.description.disabled.currently-checked-out")
+              .format(targetBranchName));
+
+    } else {
+      presentation.setDescription(
+          getNonHtmlString("action.GitMachete.CheckoutSelectedAction.description.precise")
+              .format(targetBranchName));
+    }
+  }
+
+  @Override
+  @UIEffect
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    val targetBranchName = getTargetBranchName(anActionEvent);
+    if (targetBranchName == null || targetBranchName.isEmpty()) {
+      return;
+    }
+
+    val project = getProject(anActionEvent);
+    val gitRepository = getSelectedGitRepository(anActionEvent);
+
+    if (gitRepository != null) {
+
+      log().debug(() -> "Queuing '${targetBranchName}' branch checkout background task");
+      new Task.Backgroundable(project, getString("action.GitMachete.CheckoutSelectedAction.task-title")) {
```

Comment:
There are more propertykeys that are to be renamed (not specific to `CheckoutSelectedAction`, but shared b/w all Checkout actions)